src/main/java/bio/terra/cli/Workspace.java

```diff
@@ -0,0 +1,212 @@
+   * @param description optional description
+   * @throws UserActionableException if there is no current workspace
+   */
+  public Workspace update(String name, String description) {
+    // call WSM to update the existing workspace object
+    WorkspaceDescription updatedWorkspace =
+        new WorkspaceManagerService().updateWorkspace(id, name, description);
+    logger.info("Updated workspace: {}", updatedWorkspace);
+
+    // convert the WSM object to a CLI object
+    Workspace workspace = new Workspace(updatedWorkspace);
+
+    // update the global context with the current workspace
+    Context.setWorkspace(workspace);
+    return workspace;
+  }
+
+  /** Delete the current workspace. */
+  public void delete() {
+    // call WSM to delete the existing workspace object
+    new WorkspaceManagerService().deleteWorkspace(id);
+    logger.info("Deleted workspace: {}", this);
+
+    // delete the pet SA credentials for the user
+    Context.requireUser().deletePetSaCredentials();
+
+    // unset the workspace in the current context
+    Context.setWorkspace(null);
+  }
+
+  /**
+   * List all workspaces that the current user has read access to.
+   *
+   * @param offset the offset to use when listing workspaces (zero to start from the beginning)
+   * @param limit the maximum number of workspaces to return
+   * @return list of workspaces
+   */
+  public static List<Workspace> list(int offset, int limit) {
+    // fetch the list of workspaces from WSM
+    List<WorkspaceDescription> listedWorkspaces =
+        new WorkspaceManagerService().listWorkspaces(offset, limit).getWorkspaces();
+
+    // convert the WSM objects to CLI objects
+    List<Workspace> workspaces = new ArrayList<>();
+    listedWorkspaces.forEach(wsmObject -> workspaces.add(new Workspace(wsmObject)));
+    return workspaces;
+  }
+
+  /**
+   * Get a resource by name.
+   *
+   * @throws UserActionableException if there is no resource with that name
+   */
+  public Resource getResource(String name) {
+    Optional<Resource> resourceOpt =
+        resources.stream().filter(resource -> resource.name.equals(name)).findFirst();
+    if (resourceOpt.isEmpty()) {
+      throw new UserActionableException("Resource not found: " + name);
+    }
+    return resourceOpt.get();
+  }
+
+  // ====================================================
+  // Property get/setters.
+  public UUID getId() {
+    return id;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public String getGoogleProjectId() {
+    return googleProjectId;
+  }
+
+  public String getServerName() {
+    return serverName;
+  }
+
+  public String getUserEmail() {
+    return userEmail;
+  }
+
+  public List<Resource> getResources() {
+    return Collections.unmodifiableList(resources);
+  }
+
+  public void setResources(List<Resource> resources) {
+    this.resources = resources;
+    Context.synchronizeToDisk();
```

Comment:
this is a little odd to me because the method signature kind of implies that I could use this to batch update the resources on a Workspace, but in fact the workspace as stored in WorkspaceManager isn't changed at all. By putting operations on our business objects, we have to be careful we know when they are/aren't talking to external services or updating the context.

Reply:
Agreed on it being unclear. This method is only called from one place, so I'll just move the method contents there instead.